### PR TITLE
Introduce class `OneOff` frequency

### PR DESCRIPTION
**Pull request by yhabteab**
This allows you to schedule a one-time task for a specific date time. The `None` repeat option of the `ScheduleElement` from ipl-web also depends on this PR.

### CODE REVIEW

File: src/OneOffFrequency.php
Comment (lippserd) [inline src/OneOffFrequency.php:8]: I would just call it `OneOff`.

File: src/OneOffFrequency.php
Comment (lippserd) [inline src/OneOffFrequency.php:10]: Please remove as it is not used here.

File: src/OneOffFrequency.php
Comment (lippserd) [inline src/OneOffFrequency.php:18]: Rename `$at` to `$dateTime`.

File: src/OneOffFrequency.php
Comment (lippserd) [inline src/OneOffFrequency.php:24]: I think it makes sense to remove `$dateTime = null` from the `isDue()` interface method. We never call it without a parameter if I'm not mistaken. Please open a new PR for that.

**Review by lippserd — CHANGES_REQUESTED** [review]

File: src/OneOffFrequency.php
Comment (yhabteab) [reply src/OneOffFrequency.php]: Prior to or after all other PRs have been merged?

Comment (lippserd) [reply src/OneOffFrequency.php]: We can do that afterwards.

**Comment by yhabteab** [comment]
### Blocked by

- https://github.com/Icinga/ipl-scheduler/pull/21

File: src/OneOff.php
Comment (lippserd) [inline src/OneOff.php:11]: You only need a single property.

File: tests/OneOffTest.php
Comment (lippserd) [inline tests/OneOffTest.php:14]: Also call the variable `$oneOff`.

File: tests/OneOffTest.php
Comment (lippserd) [inline tests/OneOffTest.php:9]: Also test `isExpired()` please.

File: tests/SchedulerTest.php
Comment (lippserd) [inline tests/SchedulerTest.php:285]: `testOneOffTasksRunOnlyOnce()`.

**Review by lippserd — CHANGES_REQUESTED** [review]

File: tests/OneOffTest.php
Comment (lippserd) [inline tests/OneOffTest.php:26]: `clone $now` instead of `new DateTime()` please.

File: src/OneOff.php
Comment (lippserd) [inline src/OneOff.php:20]: Should be `$dateTime >= $this->dateTime`. I think it makes sense to additionally check if the frequency is also due for something like time plus 1 hour. Should then also be done for the `Cron` class.

In `isExpired()` and `isDue()` it would also be useful to have recurring `if` patterns of the same style. Either we always put the parameter first or the property, ie.

`$this->start <= $due` vs `$due > $this->start`.

Comment (yhabteab) [reply src/OneOff.php]: > Should be `$dateTime >= $this->dateTime`. I think it makes sense to additionally check if the frequency is also due for something like time plus 1 hour. Should then also be done for the `Cron` class.

`isExpired()` returns `true` if `$dateTime` is greater than `$this->dateTime`, so I can additionally check here if it isn't the same as (**doesn't compare strictly**) `$this->dateTime`. When it isn't, `isDue()` always returns `false` because the frequency has already expired.

**Review by lippserd — APPROVED** [review]

**Pull request merged**
